codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -296,6 +297,20 @@ public Configuration configuration() {
         return frameWriter.configuration();
     }
 
+    private Http2Stream requireStream(int streamId) {
+        Http2Stream stream = connection.stream(streamId);
+        if (stream == null) {
+            String message;
```

Comment:
Compiler will ensure it is initialized before it is used if you declare it as final and initialize it in a conditional block following.